stdlib/public/core/String.swift

```diff
@@ -484,14 +484,14 @@ extension String {
     ) throws -> Int
   ) rethrows {
     self = try String(
-      uninitializedCapacity: capacity,
+      _uninitializedCapacity: capacity,
       initializingUTF8With: initializer
     )
   }
   
   @inline(__always)
   internal init(
-    uninitializedCapacity capacity: Int,
+    _uninitializedCapacity capacity: Int,
```

Comment:
LGTM, this seems like it will be a pattern we follow when we introduce new, availability-guarded public functionality that we also want to use internally.